ui/components/ui/token-input/token-input.component.js

```diff
@@ -78,6 +78,26 @@ export default class TokenInput extends PureComponent {
   handleChange = (decimalValue) => {
     const { token: { decimals } = {}, onChange } = this.props;
 
+    const newDecimalValue = decimalValue;
```

Comment:
There is a lot of identical code in `handleChange` and `handleBlur`. Perhaps `handleChange` can accept a new property like `applyDecimals` and then `handleBlur` can just call `handleChange`, but pass that parameter in